packages/queued-request-controller/tests/QueuedRequestMiddleware.test.ts

```diff
@@ -0,0 +1,425 @@
+    const middleware = createQueuedRequestMiddleware({
+      messenger,
+      useRequestQueue: () => false,
+    });
+    const req = {
+      id: '123',
+      jsonrpc: '2.0',
+      method: 'anything',
+      origin: 'anything',
+    };
+
+    await expect(
+      () =>
+        new Promise((resolve, reject) =>
+          middleware(req as any, {} as any, resolve, reject),
+        ),
+    ).rejects.toThrow("Request object is lacking a 'networkClientId'");
+  });
+
+  it('should not enqueue the request when useRequestQueue is false', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware({
+      messenger,
+      useRequestQueue: () => false,
+    });
+    const mocks = buildMocks(messenger);
+
+    await new Promise((resolve, reject) =>
+      middleware({ ...requestDefaults }, {} as any, resolve, reject),
+    );
+
+    expect(mocks.enqueueRequest).not.toHaveBeenCalled();
+  });
+
+  it('should not enqueue the request when there is no confirmation', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware({
+      messenger,
+      useRequestQueue: () => true,
+    });
+    const mocks = buildMocks(messenger);
+
+    const req = {
+      ...requestDefaults,
+      method: 'eth_chainId',
+    };
+
+    await new Promise((resolve, reject) =>
+      middleware(req, {} as any, resolve, reject),
+    );
+
+    expect(mocks.enqueueRequest).not.toHaveBeenCalled();
+  });
+
+  describe('confirmations', () => {
+    it('should resolve requests that require confirmations for infura networks', async () => {
+      const messenger = buildMessenger();
+      const middleware = createQueuedRequestMiddleware({
+        messenger,
+        useRequestQueue: () => true,
+      });
+      const mocks = buildMocks(messenger);
+
+      const req = {
+        ...requestDefaults,
+        method: 'eth_sendTransaction',
+      };
+
+      await new Promise((resolve, reject) =>
+        middleware(req, {} as any, resolve, reject),
+      );
+
+      expect(mocks.enqueueRequest).toHaveBeenCalled();
+      // infura networks do not use getNetworkClientById
+      expect(mocks.getNetworkClientById).not.toHaveBeenCalled();
+    });
+
+    it('should resolve requests that require confirmations for custom networks', async () => {
+      const messenger = buildMessenger();
+      const middleware = createQueuedRequestMiddleware({
+        messenger,
+        useRequestQueue: () => true,
+      });
+      const mocks = buildMocks(messenger);
+
+      const req = {
+        ...requestDefaults,
+        networkClientId: 'custom-rpc.com',
+        method: 'eth_sendTransaction',
+      };
+
+      await new Promise((resolve, reject) =>
+        middleware(req, {} as any, resolve, reject),
+      );
+
+      expect(mocks.enqueueRequest).toHaveBeenCalled();
+      // custom networks use getNetworkClientyId
+      expect(mocks.getNetworkClientById).toHaveBeenCalled();
```

Comment:
Is it important we know what these methods are called with rather than just the fact that they're called? I feel like that would help to tell more of the story as to what exactly this middleware does.

Reply:
I mean its not likely to catch any errors since its always just passed the request origin (if called, which is more what this test is about, is capturing the condition for when this getNetworkClientById is called vs. not called.). but it also doesn't really hurt so Ill add it.

Reply:
okay updated!